dwave/optimization/model.pxd

```diff
@@ -141,13 +142,13 @@ cdef class Symbol:
 # extension base class, so to support that we assume all ArraySymbols are
 # also Symbols (probably a fair assumption)
 cdef class ArraySymbol(Symbol):
-    # Inheriting nodes must call this method from their __init__()
-    cdef void initialize_array(self, cppArray* array_ptr) noexcept
+    # Inheriting symbols must call this method from their __init__()
+    cdef void initialize_arraynode(self, Model model, cppArrayNode* array_ptr) noexcept
 
     # Hold Array* pointer. Again this is redundant, because we're also holding
     # a pointer to Node* and we can theoretically dynamic cast each time.
     # But again, it's cheap and it simplifies things.
-    cdef cppArray* array_ptr
+    cdef cppArrayNode* array_ptr
```

Comment:
Should this have a different name to distinguish it from `cppArray*`?

Reply:
Yeah, I thought about it, and erred on the side of brevity/readability. Also a lot fewer lines to change. At the Cython level I do think `ArrayNode` is more-or-less synonymous with `Array`.